src/main/java/Main.java

```diff
@@ -0,0 +1,174 @@
+        }
+
+        try {
+            factory = CommonFactory.createFromArguments(args);
+            resources.add(new Resources("factory", factory::close));
+        } catch (Exception e) {
+            factory = new CommonFactory();
+            log.error("Failed to create common factory from args", e);
+        }
+
+        JsonMapper mapper = JsonMapper.builder().build();
+
+        Settings settings = factory.getCustomConfiguration(Settings.class, mapper);
+        MessageRouter<EventBatch> eventRouter = factory.getEventBatchRouter();
+        MessageRouter<MessageGroupBatch> messageRouter = factory.getMessageRouterMessageGroupBatch();
+        GrpcRouter grpcRouter = factory.getGrpcRouter();
+
+        run(settings, messageRouter, eventRouter, grpcRouter, resources);
+
+    }
+
+    public static void run(Settings settings, MessageRouter<MessageGroupBatch> messageRouter, MessageRouter<EventBatch> eventRouter,
+                           GrpcRouter grpcRouter, ConcurrentLinkedDeque<Resources> resources) throws ConfigError, FieldConvertError, JMException {
+
+        SessionFixBean sessionFixBean = new SessionFixBean();
+        sessionFixBean.createConfig();    //create config file for FIX
+        String sessionAlias = new SessionFixBean().getSessionAlias();
+
+        ConnectionID connectionID = ConnectionID.newBuilder().setSessionAlias(sessionAlias).build();
+
+        Event curEvent = MessageRouterUtils.storeEvent(eventRouter, Event.start(), null);
+        curEvent.name("FIX client " + sessionAlias + Instant.now());
+        curEvent.type("Microservice");
+        String rootEventId = curEvent.getId();
+
+        FixClient fixClient = new FixClient(new SessionSettings("src/main/resources/acceptor/acceptor.cfg"),
+                messageRouter, eventRouter, connectionID, rootEventId);
+
+        resources.add(new Resources("client", fixClient::stop));
+
+        ClientController controller = new ClientController(fixClient);
+
+        MessageListener<MessageGroupBatch> listener = (consumerTag, groupBatch) -> {
+            if (!controller.isRunning) controller.start(settings.autoStopAfter);
+
+            groupBatch.getGroupsList().forEach((group) -> {
+                try {
+                    if (group.getMessagesCount() != 1) {
+                        log.error("Message group contains more than 1 message", new IllegalArgumentException());
+                        throw new IllegalArgumentException();
+                    }
+                    AnyMessage message = group.getMessagesList().get(0);
+                    if (!message.hasRawMessage()) {
+                        log.error("Message in the group is not a raw message", new IllegalArgumentException());
+                        throw new IllegalArgumentException();
+                    }
+
+                    Session session = fixClient.getApplication().getSession();
```

Comment:
We should obtain session based on `SenderCompID`/`TargetCompID` fields of a message we're attempting to send. If it doesn't have them we should use the ones from a default session